docs/custom-flows/manage-sms-based-mfa.mdx

```diff
@@ -0,0 +1,374 @@
+---
+description: Learn how to use Clerk's API to build a custom flow for managing SMS-based multi-factor authentication.
+---
+
+# Build a custom flow for managing SMS-based multi-factor authentication
+
+> [!WARNING]
+> This guide is for users who want to build a *custom* user interface using the Clerk API. To use a *prebuilt* UI, you should use Clerk's [Account Portal pages](/docs/account-portal/overview) or [prebuilt components](/docs/components/overview).
+
+[Multi-factor verification (MFA)](/docs/authentication/configuration/sign-up-sign-in-options) is an added layer of security that requires users to provide a second verification factor to access an account.
+
+One of the options that Clerk supports for MFA is **SMS verification codes**. This guide will walk you through how to build a custom flow that allows users to manage their TOTP settings.
+
+> [!TIP]
+> To learn how to build a custom flow for managing authenticator app (TOTP) MFA, see the [dedicated guide](/docs/custom-flows/manage-totp-mfa).
+
+<Steps>
+
+### Enable multi-factor authentication
+
+For your users to be able to enable MFA for their account, you need to enable MFA as an MFA authentication strategy in your Clerk application.
+
+1. Navigate to the [Clerk Dashboard.](https://dashboard.clerk.com/last-active?path=user-authentication/multi-factor)
+2. In the navigation sidebar, select **User & Authentication > Multi-factor**.
+3. Enable **SMS verification code** and **Backup codes** and select **Save**.
+
+### Create the multi-factor management flow
+
+This example is written for Next.js App Router but it can be adapted for any React meta framework, such as Remix or Gatsby.
+
+This example consists of two pages:
+- The main page where users can manage their SMS MFA settings.
+- The page where users can add a phone number to their account.
+
+Use the following tabs to view the code necessary for each page.
+
+<Tabs items={["Manage MFA page", "Add phone number page"]}>
+  <Tab>
+  ```jsx {{ filename: '/app/account/manage-mfa/page.tsx' }}
+  'use client';
+
+  import * as React from 'react';
+  import { useUser } from '@clerk/nextjs';
+  import { BackupCodeResource, PhoneNumberResource } from '@clerk/types';
+  import Link from 'next/link';
+
+  // Display phone numbers reserved for MFA
+  const ManageMfaPhoneNumbers = () => {
+    const { user } = useUser();
+
+    if (!user) return null;
+
+    // Check if any phone numbers are reserved for MFA
+    const mfaPhones = user.phoneNumbers
+      .filter((ph) => ph.verification.status === 'verified')
+      .filter((ph) => ph.reservedForSecondFactor)
+      .sort((ph: PhoneNumberResource) => (ph.defaultSecondFactor ? -1 : 1));
+
+    return (
+      <>
+        <h2>Phone numbers reserved for MFA</h2>
+        {mfaPhones.length === 0 ? (
```

Comment:
I don't like ternaries used this way. I would prefer this get reverted to a separate return for the 'no number' state, like:
```
if (mfaPhones.length === 0) {
  return <p>No phone numbers configured for MFA</p>;
}

Reply:
I liked having them grouped under their corresponding H2's. but I will move them 🤷‍♀️